src/Microsoft.Identity.Client/Instance/Discovery/NetworkCacheMetadataProvider.cs

```diff
@@ -2,17 +2,21 @@
 // Licensed under the MIT License.
 
 using System.Collections.Concurrent;
+using Microsoft.Identity.Client.Core;
 
 namespace Microsoft.Identity.Client.Instance.Discovery
 {
-    internal class StaticMetadataProvider : IStaticMetadataProvider
+    internal class NetworkCacheMetadataProvider : INetworkCacheMetadataProvider
     {
         private static readonly ConcurrentDictionary<string, InstanceDiscoveryMetadataEntry> s_cache =
              new ConcurrentDictionary<string, InstanceDiscoveryMetadataEntry>();
 
-        public InstanceDiscoveryMetadataEntry GetMetadata(string environment)
+        public InstanceDiscoveryMetadataEntry GetMetadata(string environment, ICoreLogger logger)
         {
             s_cache.TryGetValue(environment, out InstanceDiscoveryMetadataEntry entry);
+            logger.Verbose($"[Instance Discovery] Tried to use network cache provider " +
```

Comment:
the + here is going to allocate 3 strings and combine them together under the covers.  this gets expensive and it's only for code readability here.  i believe you can use \ or something to define the single string constant around wrapping lines instead of doing a runtime concat of strings.  please fix this in this and other cases of logging in this PR.